PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/AzureRunProfileState.java

```diff
@@ -57,22 +60,39 @@ public ExecutionResult execute(Executor executor, @NotNull ProgramRunner program
         processHandler.startNotify();
         consoleView.attachToProcess(processHandler);
         Map<String, String> telemetryMap = new HashMap<>();
+        final Operation operation = createOperation();
         Observable.fromCallable(
             () -> {
+                if (operation != null) {
+                    operation.start();
+                    EventUtil.logEvent(EventType.info, operation, telemetryMap);
+                }
                 return this.executeSteps(processHandler, telemetryMap);
             }).subscribeOn(SchedulerProviderFactory.getInstance().getSchedulerProvider().io()).subscribe(
             (res) -> {
+                if (operation != null) {
+                    operation.complete();
+                }
                 this.sendTelemetry(telemetryMap, true, null);
                 this.onSuccess(res, processHandler);
             },
             (err) -> {
                 err.printStackTrace();
+                if (operation != null) {
+                    EventUtil.logError(operation, ErrorType.userError, new Exception(err.getMessage(), err),
+                        telemetryMap, null);
+                    operation.complete();
+                }
                 this.onFail(err.getMessage(), processHandler);
                 this.sendTelemetry(telemetryMap, false, err.getMessage());
             });
         return new DefaultExecutionResult(consoleView, processHandler);
     }
 
+    protected Operation createOperation() {
+        return null;
```

Comment:
We may return an EmptyOperationLater. which will do nothing when `start()`,`complete()`

Reply:
Good suggestion, will provide EmptyOperation later